src/sync/mod.rs

```diff
@@ -305,19 +306,25 @@ where
             // If the transaction previous to current one is committed, we found the first
             // uncommitted transaction (we've checked that the last one was not committed,
             // so scenario when all the transactions are committed is not possible).
-            if self.transaction_is_committed(previous_tx_id)? {
+            status = self.transaction_status(previous_tx_id)?;
+            if !status.is_unknown() {
                 log::trace!("Found committed transaction");
                 // Note that we were checking the previous transaction to be committed, so
                 // we return this transaction as the first not committed.
                 return Ok(Some((transaction, index)));
             }
         }
-        // If we reach this branch then the transaction previous to the first one was not
-        // committed, but previous transaction for the first anchoring transaction always
-        // is funding. Therefore, the first funding transaction has no confirmation.
-        Err(SyncWithBitcoinError::UnconfirmedFundingTransaction(
-            transaction.prev_tx_id(),
-        ))
+
+        if status.confirmations().is_none() {
```

Comment:
Maybe make the `Ok` case the first one and add a comment describing why it's `Ok(None)`?

Reply:
And also add a `log::trace` entries.